crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -78,50 +74,87 @@ pub struct Mesh {
+/// other APIs can have other conventions, `OpenGL` starts at bottom-left.
+/// - It is possible and sometimes useful for multiple vertices to have the same
+/// [position attribute](Mesh::ATTRIBUTE_POSITION) value,
+/// it's a common technique in 3D modelling for complex UV mapping or other calculations.
+///
+/// ## Use with `StandardMaterial`
+///
+/// To render correctly with `StandardMaterial`, a mesh needs to have properly defined:
+/// - [`UVs`](Mesh::ATTRIBUTE_UV_0): Bevy needs to know how to map a texture onto the mesh
+/// (also true for `ColorMaterial`).
+/// - [`Normals`](Mesh::ATTRIBUTE_NORMAL): Bevy needs to know how light interacts with your mesh.
+/// [0.0, 0.0, 1.0] is very common for simple flat meshes on the XY plane,
+/// because simple meshes are smooth and they don't require complex light calculations.
+/// - Vertex winding order: by default, `StandardMaterial.cull_mode` is [`Some(Face::Front)`](crate::render_resource::Face),
+/// which means that Bevy would *only* render the "front" of each triangle, which
+/// is the side of the triangle from where the vertices appear in a *counter-clockwise* order.
+///
+// TODO: allow values to be unloaded after been submitting to the GPU to conserve memory
+#[derive(Debug, TypeUuid, TypePath, Clone)]
+#[uuid = "8ecbac0f-f545-4473-ad43-e1f4243af51e"]
+pub struct Mesh {
+    primitive_topology: PrimitiveTopology,
+    /// `std::collections::BTreeMap` with all defined vertex attributes (Positions, Normals, ...)
+    /// for this mesh. Attribute ids to attribute values.
+    /// Uses a BTreeMap because, unlike HashMap, it has a defined iteration order,
+    /// which allows easy stable VertexBuffers (i.e. same buffer order)
+    attributes: BTreeMap<MeshVertexAttributeId, MeshAttributeData>,
+    indices: Option<Indices>,
+    morph_targets: Option<Handle<Image>>,
+    morph_target_names: Option<Vec<String>>,
+}
+
 impl Mesh {
-    /// Where the vertex is located in space. Use in conjunction with [`Mesh::insert_attribute`]
+    /// Where the vertex is located in space. Use in conjunction with [`Mesh::insert_attribute`].
     pub const ATTRIBUTE_POSITION: MeshVertexAttribute =
         MeshVertexAttribute::new("Vertex_Position", 0, VertexFormat::Float32x3);
 
     /// The direction the vertex normal is facing in.
-    /// Use in conjunction with [`Mesh::insert_attribute`]
+    /// Use in conjunction with [`Mesh::insert_attribute`].
     pub const ATTRIBUTE_NORMAL: MeshVertexAttribute =
         MeshVertexAttribute::new("Vertex_Normal", 1, VertexFormat::Float32x3);
 
-    /// Texture coordinates for the vertex. Use in conjunction with [`Mesh::insert_attribute`]
+    /// Texture coordinates for the vertex. Use in conjunction with [`Mesh::insert_attribute`].
+    ///
+    /// Values are generally between 0. and 1., `[0.,0.]` being the top left of the texture,
+    /// and [1.,1.] the bottom-right. Values outside this range will generate visual artifacts with
```

Comment:
I believe this should read "Other values will be clamped to this range by default".

Reply:
Well they're not clamped it makes an artifact. it "expands" the texture with the color value it has at the closest border

If I put "2.0" instead of "1.0" for y UV values for example:
![image](https://github.com/bevyengine/bevy/assets/134181069/e8339c06-1579-4833-bc88-93b2ca944d8e)

I'll fix the typo and mistake you mentioned earlier, thank you!

Reply:
Yes, this is the behaviour you would expect if the texture coordinates are clamped to [0, 1]. The value of the texel at the closest edge of the texture will be used.

This behaviour is specified by WebGPU - https://gpuweb.github.io/gpuweb/#enumdef-gpuaddressmode
It just happens to result in that artifact with the texture you chose, but it might produce reasonable results for other cases.

Reply:
Okay but it's clamped by the rendering process, for all the pixels with outside value, not for each vertex, otherwise putting `2.` would have the same effect as putting `1.`.

In general, for the end user:
- this is something they don't want
- it will be perceived as a visual artifact
- if we tell them the value is clamped they might get the wrong impression that it doesn't matter if they put values outside the range

I think going into details is more confusing than just saying "don't do it" here.

Reply:
I believe the main use of this behaviour is producing a solid border color. For example, the following logo has a solid red border, and sampling it with uv coordinates outside the 0-1 range can be used to extend this border to cover a surface that is not the same shape without warping the logo:

<img width="474" alt="kfc logo" src="https://github.com/bevyengine/bevy/assets/9408210/7fd6558c-4c8d-4486-9f28-ddb40f1a9a4b">

But I'm happy to defer to other reviewers here if the consensus is that it confuses the end user.

Reply:
Fair enough, I'll try to change the formulation to explain this.

Reply:
looks good to me